core/test/solver/bicg.cpp

```diff
@@ -185,7 +185,12 @@ TYPED_TEST(Bicg, CanSetPreconditionerGenerator)
                     .with_reduction_factor(
                         gko::remove_complex<value_type>(1e-6))
                     .on(this->exec))
-            .with_preconditioner(Solver::build().on(this->exec))
+            .with_preconditioner(
+                Solver::build()
+                    .with_criteria(
+                        gko::stop::Iteration::build().with_max_iters(3u).on(
```

Comment:
I do understand correctly that you only run 3 iterations?

Reply:
That's a core test, it doesn't actually run the kernels, it just checks the overall framework/setup.

Reply:
After this change, non-setting criteria throws `NotSupported`. Need to add a valid criteria to run the test.